pySDC/implementations/problem_classes/LeakySuperconductor.py

```diff
@@ -0,0 +1,281 @@
+
+    def eval_f_non_linear(self, u, t):
+        """
+        Get the non-linear part of f
+
+        Args:
+            u (dtype_u): current values
+            t (float): current time
+
+        Returns:
+            dtype_u: the non-linear part of the RHS
+        """
+        u_thresh = self.params.u_thresh
+        u_max = self.params.u_max
+        Q_max = self.params.Q_max
+        me = self.dtype_u(self.init)
+
+        me[:] = (u - u_thresh) / (u_max - u_thresh) * Q_max
+        me[u < u_thresh] = 0
+        me[self.leak] = Q_max
+        me[u >= u_max] = Q_max
+
+        # boundary conditions
+        me[0] = 0.0
+        me[-1] = 0.0
+
+        return me
+
+    def eval_f(self, u, t):
+        """
+        Evaluate the full right hand side.
+
+        Args:
+            u (dtype_u): Current solution
+            t (float): Current time
+
+        Returns:
+            dtype_f: The right hand side
+        """
+        f = self.dtype_f(self.init)
+        f[:] = (self.A.dot(u.flatten()).reshape(self.params.nvars) + self.eval_f_non_linear(u, t)) / self.params.Cv
+        return f
+
+    def solve_system(self, rhs, factor, u0, t):
+        """
+        Simple linear solver for (I-factor*A)u = rhs
+
+        Args:
+            rhs (dtype_f): right-hand side for the linear system
+            factor (float): abbrev. for the local stepsize (or any other factor required)
+            u0 (dtype_u): initial guess for the iterative solver
+            t (float): current time (e.g. for time-dependent BCs)
+
+        Returns:
+            dtype_u: solution as mesh
+        """
+
+        def get_non_linear_Jacobian(u):
+            """
+            Evaluate the non-linear part of the Jacobian only
+
+            Args:
+                u (dtype_u): Current solution
+
+            Returns:
+                dtype_u: The derivative of the non-linear part of the solution w.r.t. to the solution.
+            """
+            u_thresh = self.params.u_thresh
+            u_max = self.params.u_max
+            Q_max = self.params.Q_max
+            me = self.dtype_u(self.init)
+
+            me[:] = Q_max / (u_max - u_thresh)
+            me[u < u_thresh] = 0
+            me[u > u_max] = 0
+            me[self.leak] = 0
+
+            # boundary conditions
+            me[0] = 0.0
+            me[-1] = 0.0
+
+            me = self.dtype_u(self.init)
+            return me
+
+        u = self.dtype_u(u0)
+        res = np.inf
+        for _n in range(0, self.params.liniter):
+            # assemble G such that G(u) = 0 at the solution of the step
+            G = u - factor * self.eval_f(u, t) - rhs
+
+            res = np.linalg.norm(G, np.inf)
+            if res <= self.params.lintol or np.isnan(res):
+                break
+
+            # assemble Jacobian J of G
+            J = self.Id - factor * (self.A + get_non_linear_Jacobian(u))
+
+            # solve the linear system
+            delta = np.linalg.solve(J, G)
```

Comment:
Could make this iterative, too, using the flag you already have..

Reply:
I used zeros as initial guess because I am computing a correction here, which should be small. Does that make sense? There is no information about the spatial profile in zeros left, but I couldn't think of anything better. Suggestions?

Reply:
No, makes sense. You want the delta to be zero at some point, so this is meaningful.